odl/discr/grid.py

```diff
@@ -72,34 +73,19 @@ def sparse_meshgrid(*x):
     return tuple(mesh)
 
 
-class TensorGrid(Set):
+class RectGrid(Set):
 
-    """An n-dimensional tensor grid.
+    """An n-dimensional rectilinear grid.
 
-    A tensor grid is the set of points defined by all possible
+    A rectilinear grid is the set of points defined by all possible
     combination of coordinates taken from fixed coordinate vectors.
 
-    In 2 dimensions, for example, given two coordinate vectors::
-
-        coord_vec1 = [0, 1]
-        coord_vec2 = [-1, 0, 2]
-
-    the corresponding tensor grid is the set of all 2d points whose
-    first component is from ``coord_vec1`` and the second component from
-    ``coord_vec2``. The total number of such points is 2 * 3 = 6::
-
-        points = [[0, -1], [0, 0], [0, 2],
-                  [1, -1], [1, 0], [1, 2]]
-
-    Note that this is the standard 'C' ordering where the first axis
-    (``coord_vec1``) varies slowest. Ordering is only relevant when
-    the point array is actually created; the grid itself is independent
-    of this ordering.
-
-    The storage need for a tensor grid is only the sum of the lengths
+    The storage need for a rectilinear grid is only the sum of the lengths
     of the coordinate vectors, while the total number of points is
-    the product of these lengths. This class makes use of this
-    sparse storage.
+    the product of these lengths. This class makes use of that
+    sparse storage scheme.
+
+    See ``Notes`` for details.
```

Comment:
That may be true, but *how often* is that important compared to the parameter explanations? 

The math notes are important to understand what this represents, and that's usually when you read the doc for the first time. After that, you perhaps occasionally check the math but not often.

On the other hand, the parameters are important on a day-to-day basis when you use the function or class in practice. Of course the signature is printed first (usually, sometimes it doesn't work but that's nothing to count in), but when there are `kwargs` or if you need details about the parameters you need that section. And then at least I find it annoying if there are too many other things in the way that make me scroll every time I need info about the arguments.

So even from a dev perspective I feel that in the long run, parameters are more important than other details in the notes.



> other IDEs may do this "properly" (i.e. distinguish between a class and its initializer in a better way than spyder)

Not sure what a better way would look like, or how other IDEs handle this. To me it seems reasonable to show the class docstring and the initialization doc if a user asks for documentation on a class.
The alternative would be to let the user choose, with pop-up menus like (apparently) in PyCharm. I personally find that annoying.